src/main/webapp/js/components/settings/csv.jsx

```diff
@@ -0,0 +1,319 @@
+        entity,
+      },
+      isSelectDisabled: false,
+    }, () => {
+      if (!this.state.csv.id) {
+        this.props.createCsvConfig(this.state.csv);
+      } else {
+        this.props.saveCsvConfig(this.state.csv);
+      }
+    });
+  }
+
+  getOptions() {
+    return _.map(this.props.fieldConfig[this.state.csv.entity], v => ({
+      label: v.displayName,
+      value: v.id,
+    }));
+  }
+
+  getRelatedFieldValues(relatedField) {
+    return _.map(this.props.entity[relatedField], v => ({
+      label: v.name,
+      value: v.id,
+    }));
+  }
+
+  getFields(values, index) {
+    let hidden = true;
+    let hiddenRelated = true;
+    if (index) {
+      _.map(this.props.fieldConfig[this.state.csv.entity], (entity) => {
+        if (entity.id === values[index].fieldConfigId) {
+          if (entity.fieldType === 'DATE' || entity.fieldType === 'DATE_TIME') {
+            hidden = false;
+          }
+          if (entity.relatedEntity) {
+            hiddenRelated = false;
+          }
+        }
+      });
+    }
+
+    return [
+      {
+        name: 'fieldConfigId',
+        fieldType: 'ENUM',
+        displayName: 'Field name',
+        required: true,
+        options: this.getOptions(),
+      },
+      {
+        name: 'csvName',
+        fieldType: 'TEXT',
+        displayName: 'CSV name',
+        required: true,
+      },
+      {
+        name: 'format',
+        fieldType: 'TEXT',
+        displayName: 'Format',
+        required: true,
+        hidden,
+      },
+      {
+        name: 'defaultValue',
+        fieldType: 'TEXT',
+        displayName: 'Default value',
+        required: false,
+      },
+      {
+        name: 'csvRelatedFields',
+        displayName: '',
+        required: false,
+        hidden: hiddenRelated,
+      },
+    ];
+  }
+
+  getRelatedFields(relatedField) {
+    return [
+      {
+        name: 'relatedField',
+        fieldType: 'ENUM',
+        displayName: 'Related field',
+        required: true,
+        options: this.getRelatedFieldValues(relatedField),
+        arrayFields: true,
+      },
+      {
+        name: 'relatedFieldCsvName',
+        fieldType: 'TEXT',
+        displayName: 'Related field CSV name',
+        required: true,
+        arrayFields: true,
+      },
+    ];
+  }
+
+  validate = (values) => {
+    const errors = {};
+    errors.csvFields = [];
+
+    _.forEach(values.csvFields, (csvField, key) => {
+      _.forEach(this.getFields(values.csvFields, key), (config) => {
+        const val = csvField[config.name];
+
+        if (config.required && !config.hidden && (_.isNil(val) || val === '')) {
+          errors.csvFields[key] = { [config.name]: `${config.displayName} is required` };
+        }
+      });
+    });
+
+    return errors;
+  };
+
+  onChange = (entity) => {
+    this.setState({ csv: { ...this.state.csv, entity } }, () => this.fetchData(entity));
+  };
+
+  fetchData(entity) {
+    this.props.fetchCsvConfig(this.state.csv.entity)
```

Comment:
I don't think you should fetch the csv configs in app.jsx, like the field config is fetched